applications/GeoMechanicsApplication/custom_workflows/time_loop_executor.hpp

```diff
@@ -0,0 +1,99 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//                   Anne van de Graaf
+//                   Wijtze Pieter Kikstra
+//
+
+#pragma once
+
+#include <vector>
+#include <memory>
+
+#include "time_loop_executor_interface.h"
+#include "time_step_end_state.hpp"
+#include "time_incrementor.h"
+#include "time_step_executor.h"
+#include "strategy_wrapper.hpp"
+
+namespace Kratos
+{
+
+class Process;
+
+class TimeLoopExecutor : public TimeLoopExecutorInterface{
+public :
+    void SetProcessObservables(const std::vector<std::weak_ptr<Process>>& rProcessObservables) override
+    {
+        mTimeStepExecutor->SetProcessObservables(rProcessObservables);
+    }
+
+    void SetTimeIncrementor(std::unique_ptr<TimeIncrementor> pTimeIncrementor) override
+    {
+        mTimeIncrementor = std::move(pTimeIncrementor);
+    }
+
+    void SetSolverStrategyTimeStepExecutor(std::shared_ptr<StrategyWrapper> pStrategyWrapper) override
+    {
+        mStrategyWrapper = std::move(pStrategyWrapper);
+        mTimeStepExecutor->SetSolverStrategy(mStrategyWrapper);
+    }
+
+    std::vector<TimeStepEndState> Run(TimeStepEndState EndState) override
+    {
+        mStrategyWrapper->SaveTotalDisplacementFieldAtStartOfStage();
+        std::vector<TimeStepEndState> result;
+        while (mTimeIncrementor->WantNextStep(EndState)) {
+            mStrategyWrapper->IncrementStepNumber();
+            // clone without end time, the end time is overwritten anyway
+            mStrategyWrapper->CloneTimeStep();
+            EndState = RunCycleLoop(EndState);
+            mStrategyWrapper->AccumulateTotalDisplacementField();
+            mStrategyWrapper->FinalizeSolutionStep();
+            result.emplace_back(EndState);
+            mStrategyWrapper->OutputProcess();
```

Comment:
It may read more logically if we swap these two statements (that shouldn't have any side effects, rights?).
```suggestion
            mStrategyWrapper->OutputProcess();
            result.emplace_back(EndState);
```

Reply:
I do not expect effects of it so done.